backend/restapi/services/recommendation.py

```diff
@@ -0,0 +1,45 @@
+from openai import OpenAI
+import os
+
+
+def get_recommendations(phrases, tags):
+
+    client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))
+
+    limit = 10  # number of movies to return
+
+    example = [
+        {
+            "id": 0,
+            "title": "The Godfather",
+            "genres": ["Crime", "Drama"],
+            "year": 1972,
+            "rating": 9.2,
+            "duration": 175,
+            "description": "The aging patriarch of an organized crime dynasty transfers control of his clandestine empire to his reluctant son.",  # noqa: E501
+        },
+        {
+            "id": 1,
+            "title": "The Shawshank Redemption",
+            "genres": ["Drama"],
+            "year": 1994,
+            "rating": 9.3,
+            "duration": 142,
+            "description": "Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.",  # noqa: E501
+        },
+    ]
```

Comment:
use https://platform.openai.com/docs/guides/structured-outputs/examples#chain-of-thought, with gpt-4o-mini model